DataDog/datadog-agent
# Add ability to change log_level at runtime to system-probe

### What does this PR do?

This PR adds a `config` subcommand, like the core agent, which allows changing specific configuration values at runtime. This PR adds the ability to change the `log_level` config value for system-probe.

### Motivation

Increased need to debug system-probe on production systems without restarting.

### Additional Notes

- The runtime setting code was moved from `cmd/agent/app/settings` to `pkg/config/settings` so that other agents could re-use it.
- The system-probe is communicated to over the existing unix socket on Linux, rather than TCP like the core agent.

### Describe your test plan

1. Start system-probe
2. Get current `log_level` with `system-probe config get log_level`
3. Change `log_level` with `system-probe config set log_level debug` (or another log level value)
4. Notice the change in level via the system-probe logs.
